### ISSUE TITLE
[Feature]: WCAG 2.2 improvements for Fluent v9

### ISSUE THREAD

**Issue by gouttierre** [microsoft/fluentui#32383]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

react.fluentui.dev

### Are you reporting an Accessibility issue?

yes

### Reproduction

https://react.fluentui.dev/

## Expected Behavior
**Failing controls** 
The following controls currently fail Target Size (Minimum): 
> 📢 Designs are needed for the below changes.
- [x] #33648
  - Resizable option. This may fail even if an alternative mandated by Dragging Movements is added, if the target size of the alternative is also too narrow.
  - [Design User Story 20283 Target Size (Minimum) (AA) - Drawer](https://dev.azure.com/microsoftdesign/X3%20Design%20Horizontals/_workitems/edit/20283)
 
- [x] #33649
  - Small variant, but only if it is immediately preceding or following an interactive control like a link. 
  - [Design User Story 20287 Target Size (Minimum) (AA) - InfoLabel ](https://dev.azure.com/microsoftdesign/X3%20Design%20Horizontals/_workitems/edit/20284)

- [x] #33650
  - Both the medium and small variants fail for the menu button in the horizontal axis.  The medium variant fails by only 1px, which should be easy to fix. The small variant fails by 9px. 
  - [Design User Story 20288 Target Size (Minimum) (AA) - SplitButton](https://dev.azure.com/microsoftdesign/X3%20Design%20Horizontals/_workitems/edit/20288)

- [ ] #33652
  - Include extra small variants.
  - [Design User Story 20289 Target Size (Minimum) (AA) - SwatchPicker](https://dev.azure.com/microsoftdesign/X3%20Design%20Horizontals/_workitems/edit/20289)

- [x] #33651
  - The default size tags (small) and the extra small tags are not tall enough, and do not have spacing above or below. Clicking outside the tag focuses the input and toggles the listbox, and wrapping lines tags only have 4px of space between lines (they would need 8px to pass without changing tag size). 
  - [Design User Story 20289Target Size (Minimum) (AA) - Tag/TagPicker](https://dev.azure.com/microsoftdesign/X3%20Design%20Horizontals/_workitems/edit/20289)

 
**Passing controls (but with anticipated false positives)** 
The following controls currently do not fail this criterion, though we may get questions about them we should document these as known for transparency i.e. ([Fluent v9 known issues - Overview (azure.com)](https://dev.azure.com/microsoftdesign/fluent-ui/_wiki/wikis/fluent-ui.wiki/334/Fluent-v9-known-issues))
- [ ] **Avatar**/**AvatarGroup** – the 16px and 20px sizes currently do not fail, as there is adequate spacing around them and only the overflow button is interactive. The 16px size (and only that size) would fail if all avatars were interactive. 
- [ ] **Combobox** – the expand/collapse button does not fail, since there is an alternative way to open the combo (clicking on the input). 
- [ ] **DataGrid**/**Table** resizing (debatable) – the resize handle does not fail, since there is an equivalent way to perform the same action (the context menu + keyboard resizing). This will more clearly pass once Dragging Movements is met, since then it will be an alternative that only relies on pointer interactions. 
- [ ] **Rating** (debatable) – the half-step version fails for all sizes, and all sizes apart from extra-large fail. If this is considered a single control rather than multiple radios, it would pass if all sizes smaller than extra-large have adequate top/bottom margins. 
- [ ] **Slider** – despite being under 24px tall, it has adequate space above and below the input. 
- [ ] **SpinButton** – the up/down buttons do not fail, since there is an equivalent way to perform the same action (typing in the input). 

### Requested priority

High

### Products/sites affected

Fluent v9 

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
📢 WCAG 2.2 improvements for Fluent v9 are completed.
